pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -80,17 +78,7 @@ func (s *DiscoveryServer) generateRawClusters(node *model.Proxy, push *model.Pus
 		return nil, err
 	}
 
-	if sdsTokenPath, found := node.Metadata[model.NodeMetadataSdsTokenPath]; found && len(sdsTokenPath) > 0 &&
-		pilot.EnableCDSPrecomputation() {
-		// If SDS_TOKEN_PATH is in the node metadata, make a copy of rawClusters so that
-		// the path of SDS token will be applied to the copied clusters.
-		rawClusters = CopyClusters(rawClusters)
-	}
-
 	for _, c := range rawClusters {
-		if pilot.EnableCDSPrecomputation() {
-			SetTokenPathForSdsFromProxyMetadata(c, node)
-		}
```

Comment:
In a subsequent PR or same, can you move this sds code into cluster.go ? It would ideally be done in the auth plugin I guess.

Reply:
Yeah, good point. I'll add it to this PR.

Reply:
Sure, I'll work that into this PR.

Reply:
Isn't this logic duplicated [here](https://github.com/istio/istio/blob/master/pilot/pkg/model/authentication.go#L133-L136) without the flag and being used indirectly from cluster.go already?